### PR TITLE
Add TypeScript definitions file

### LINKED ISSUE #1: TypeScript definitions
This project just needs typings for a 1.0 release~!

### PR DESCRIPTION
**Pull request by krismuniz**
Hello there :wave:

This PR adds TypeScript definitions for `formee` and hopefully closes #1 :)

- Adds `src/index.d.ts`
- Updates package.json to include `types` property to `package.json`

P.S. This module is awesome!

Let me know if you need any changes ✏️

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+interface BindOptions {
+    onSubmit(event: Event): boolean;
+    onError(event: Event): boolean;
+    rules?: ValidationRules;
+}
+
+export declare function bind(form: HTMLFormElement, opts: BindOptions): HTMLFormElement;
+
+export {};
```
Comment (lukeed) [inline src/index.d.ts]: This part looks odd to me, what does it mean?

**Comment by lukeed** [comment]
This is awesome, thank you!

Can you move the file into root though, as `formee.d.ts`? That will keep it consistent with my other modules.

---

Pinging @marvinhagemeister for a review, but only if you have a moment! 🙏

File: src/index.d.ts (hunk 1 above)
Comment (krismuniz) [reply src/index.d.ts]: The `export {}` line, right? I honestly don't know why TS is generating it... 🤔

File: formee.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export declare function serialize(form: HTMLFormElement): object;
+
+interface ValidationRules {
```
Comment (id0Sch) [inline formee.d.ts]: mind exporting this as well? 
it's helpful when you want to have these are variables before you call the function :)

File: formee.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    [key: string]: RegExp | ((val: string, data: object) => boolean | string);
+}
+
+export declare function validate(form: HTMLFormElement, rules: ValidationRules, toCheck?: string): object;
+
+interface BindOptions {
```
Comment (id0Sch) [inline formee.d.ts]: and export this also (same as line #3 )

**Review by id0Sch — COMMENTED** [review]
left a couple of notes, hope it's okay :)

**Comment by lukeed** [comment]
@id0Sch Exported them, thanks~!

**Review by lukeed — APPROVED** [review]
Doesn't seem like that `export {}` was needed at all. Locally, there was no difference either way 😅 

Thank you again!

**Comment by krismuniz** [comment]
Awesome! No problem :)

**Pull request merged**